src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs

```diff
@@ -124,9 +124,11 @@ public bool EnsureCacheEntryIfExists(Checksum checksum)
             return _assetCache.TryGetAsset<object>(checksum, out _);
         }
 
-        public async Task SynchronizeAssetsAsync(ISet<Checksum> checksums, CancellationToken cancellationToken)
+        public async ValueTask SynchronizeAssetsAsync(ISet<Checksum> checksums, CancellationToken cancellationToken)
         {
             Debug.Assert(!checksums.Contains(Checksum.Null));
+            if (checksums.Count == 0)
+                return;
```

Comment:
this is a core change/thing i noticed.  When logging this information. we call into this tons of times with nothing to actually sync.  Roughly 98%+ of the time this happens.  so this can insta-bail.  Which allows this to sensibly be a value task.  the rest is just bubbling that upwards for all calls that themselves are just calling awaiting value-task helpers.